modules/pyramid/src/pyramid.cpp

```diff
@@ -0,0 +1,321 @@
+#include "precomp.hpp"
+
+using namespace std;
+using namespace cv;
```

Comment:
the code should not be `using name space cv; using namespace pyramid;` but should be inside cv::pyramid

like:

    namespace cv {
    namespace pyramid {
    ... // code
    }
    }

this is also, why the linker does not find `cv::pyramid::LaplacianPyramid::LaplacianPyramid(Mat &,int)` and others